ward/terminal.py

```diff
@@ -385,7 +416,39 @@ def output_all_test_results(
         output_tests = self.runtime_output_strategies.get(
             self.test_output_style, output_test_per_line
         )
-        all_results = output_tests(fail_limit, test_results_gen)
+
+        spinner = SpinnerColumn(style="pass.textonly")
+        bar = BarColumn(complete_style="pass.textonly")
+        progress = Progress(
+            spinner,
+            TimeElapsedColumn(),
+            bar,
+            "[progress.percentage]{task.percentage:>3.0f}%",
+            "[progress.percentage][{task.completed} / {task.total}]",
+            TimeRemainingColumn(),
+            console=console,
+            transient=True,
+            auto_refresh=True,
+            # TODO: figure out how to make the progress bar work with printing with end="" (for dots)
+            disable=self.test_output_style != "test-per-line",
+        )
+
+        task = progress.add_task(
+            "Testing...", total=self.suite.num_tests_with_parameterization
+        )
+
+        def test_done() -> None:
+            progress.update(task, advance=1, refresh=True)
```

Comment:
I'm guessing this is done after each test?

You probably don't want to do `refresh=True` here.  The progress bar will update at a steady frame rate when auto refresh is enabled.

Reply:
That helped! Still some flickering, but much improved, and it sounds like it's probably my terminal (Windows Terminal) that's the underlying cause.

Thanks for stopping by @willmcgugan !